src/test/PoolSwapTest.sol

```diff
@@ -52,53 +53,50 @@ contract PoolSwapTest is PoolTestBase {
 
         CallbackData memory data = abi.decode(rawData, (CallbackData));
 
-        (, uint256 poolBalanceBefore0, int256 deltaBefore0) =
-            _fetchBalances(data.key.currency0, data.sender, address(this));
-        (, uint256 poolBalanceBefore1, int256 deltaBefore1) =
-            _fetchBalances(data.key.currency1, data.sender, address(this));
+        (,, int256 deltaBefore0) = _fetchBalances(data.key.currency0, data.sender, address(this));
+        (,, int256 deltaBefore1) = _fetchBalances(data.key.currency1, data.sender, address(this));
 
         require(deltaBefore0 == 0, "deltaBefore0 is not equal to 0");
         require(deltaBefore1 == 0, "deltaBefore1 is not equal to 0");
 
         BalanceDelta delta = manager.swap(data.key, data.params, data.hookData);
 
-        (, uint256 poolBalanceAfter0, int256 deltaAfter0) =
-            _fetchBalances(data.key.currency0, data.sender, address(this));
-        (, uint256 poolBalanceAfter1, int256 deltaAfter1) =
-            _fetchBalances(data.key.currency1, data.sender, address(this));
-
-        require(poolBalanceBefore0 == poolBalanceAfter0, "poolBalanceBefore0 is not equal to poolBalanceAfter0");
-        require(poolBalanceBefore1 == poolBalanceAfter1, "poolBalanceBefore1 is not equal to poolBalanceAfter1");
+        (,, int256 deltaAfter0) = _fetchBalances(data.key.currency0, data.sender, address(this));
+        (,, int256 deltaAfter1) = _fetchBalances(data.key.currency1, data.sender, address(this));
 
         if (data.params.zeroForOne) {
             if (data.params.amountSpecified < 0) {
                 // exact input, 0 for 1
                 require(
-                    deltaAfter0 == data.params.amountSpecified,
+                    deltaAfter0 >= data.params.amountSpecified,
```

Comment:
if zero for one is true and it is an exact Input swap, meaning currency0 is specified why is deltaAfter0 allowed to be > than the amountSpecified. I thought this was one of the properties we wanted to maintain.

Also isn't it not possible to make this happen?

ie swapper inputs -10 to swap. if a hook returns a negative delta (say -1) the swapAmount becomes -11 but the user still only owes -10

& in the other case swapper inputs -10. if a hook returns a positive delta (say +1) the swapAmount becomes -9 but the user still owes -10.

in what case is the users delta in the currency specified not exactly equal to the amount specified?

Reply:
the extra cases are all for when the pool is low on liquidity. Even without hooks returning deltas, the user might not get amountSpecified is theyre trading on a pool that doesnt have enough liquidity. Its just that we dont have a swap test for that right now.
If you send `amountSpecified = -10` but the pool only has enough liquidity to take 5 input for the output thats left. Then the user ends up with a delta of `-5` because of that. So thats why this is actually a `>=` not a `==`.

That is additionally emphasised by a hook returning a delta.
With the same example above is a hook additionally returns `-1` and the swapAmount becomes `-11`. The pool swaps `5`, 1 of which was from the hook. So the user ends up with a delta of `-5--1` (which is the actual swap delta, minus the hook's delta), = `-4`

Reply:
Makes sense - I think for some reason I thought that even on low liquidity, it would trade your full delta, & not do a partial amount but I realize that is definitely not the case

Reply:
We should definitely have low liquidity tests (even outside of hooks return deltas, because I realize that this assertion is _incorrect_ outside of this pr).